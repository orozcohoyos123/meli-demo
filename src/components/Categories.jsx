import React from "react";
import PropTypes from "prop-types";
import "./styles/Categories.scss";
import NextLogo from "../assets/ic_next.png";
import { Link } from "react-router-dom";

/**
 * Component for showing the categories from the products
 * 
 * @component
 
 * @prop {Object[]} items - Categories' list
 * @prop {string} items[].id -  Category's id
 * @prop {string} items[].name - Category's name
 * @example
 *  const items = 
 * [
 *  {id: "MLA1144", name: "Consolas y Videojuegos"},
 *  {id: "MLA438566", name: "Consolas"}
 * ]
 * return (<Categories items={items} />)
 */
const Categories = (props) => {
   return (
    <div className="categories col-11">
      <div className="categories__container">
        {props.items && props.items.map((item, index) => {
          const lastItem = props.items.length - 1 === index;
          return (
            <React.Fragment key={`${item.id}_${index}`}>
              <Link to={`/items?search=${item.name}`}>
                <span
                  className={`categories__item ${lastItem ? "bold" : ""}`}
                >
                  {item.name}
                </span>
              </Link>
              {!lastItem && (
                <img 
                  src={NextLogo}
                  className="categories__separator"
                  alt="Next"
                />
              )}
            </React.Fragment>
          );
        })}
      </div>
    </div>
  );
};

Categories.propTypes = {
  /**
   * Categories' array
   */
  categories: PropTypes.array
};


export default Categories;
