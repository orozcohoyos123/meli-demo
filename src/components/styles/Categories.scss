@import '../../styles/CommonColors.scss';
@import '../../styles/CommonFontSize.scss';

.bold {
  color: $gray;
  font-weight: bold;
}

.categories {
  padding: 16px 0;
}

.categories__item {
  color: $middle-gray;
  cursor: pointer;
  font-size: $small;
}

.categories__separator {
  height: 14px;
  margin: 0 5px 2px 5px;
}
