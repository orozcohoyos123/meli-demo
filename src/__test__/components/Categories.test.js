import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer'
import Categories from '../../components/Categories';
import { MemoryRouter } from 'react-router-dom';

const mockCategories = [
    {id: "MLA1144", name: "Consolas y Videojuegos"},
    {id: "MLA438566", name: "Consolas"}
]

describe('<Categories />', () => {
    test('Debe renderizar el componente Categories con mock de categorias', () => {
        const categories = mount(
            <MemoryRouter>
                <Categories items={mockCategories} />
            </MemoryRouter>
        );

        const items = categories.find('.categories__item');
        expect(categories.length).toEqual(1);
        expect(items.length).toEqual(mockCategories.length);
    })

    test('Deben haber cero items si no se envÃ­a el mock de categorias', () => {
        const categories = mount(
            <MemoryRouter>
                <Categories />
            </MemoryRouter>
        );

        const items = categories.find('.categories__item');
        expect(items.length).toEqual(0);
    })
});

describe('Categories sanpshot', () => {
    test('Comprobar UI componente Categories', () => {
        const categories = 
            create(
                <MemoryRouter>
                    <Categories items={mockCategories} />
                </MemoryRouter>
            );
        
        expect(categories.toJSON()).toMatchSnapshot();
    })
});
