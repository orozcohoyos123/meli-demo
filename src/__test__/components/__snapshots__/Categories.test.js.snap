// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories sanpshot Comprobar UI componente Categories 1`] = `
<div
  className="categories col-11"
>
  <div
    className="categories__container"
  >
    <a
      href="/items?search=Consolas y Videojuegos"
      onClick={[Function]}
    >
      <span
        className="categories__item "
      >
        Consolas y Videojuegos
      </span>
    </a>
    <img
      alt="Next"
      className="categories__separator"
      src="ic_next.png"
    />
    <a
      href="/items?search=Consolas"
      onClick={[Function]}
    >
      <span
        className="categories__item bold"
      >
        Consolas
      </span>
    </a>
  </div>
</div>
`;
